Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.cpp

```diff
@@ -306,6 +313,66 @@ void WebPrintOperationGtk::endPrint(cairo_t* cr)
     cairo_surface_finish(cairo_get_target(cr));
     printDone(m_buffer.take(), { });
 }
+#elif USE(SKIA)
+void WebPrintOperationGtk::startPage(SkPictureRecorder& recorder)
+{
+    if (!currentPageIsFirstPageOfSheet()) {
+        ASSERT(m_pageCanvas);
+        return;
+    }
+
+    ASSERT(!m_pageCanvas);
+
+    GtkPaperSize* paperSize = gtk_page_setup_get_paper_size(m_pageSetup.get());
+    double width = gtk_paper_size_get_width(paperSize, GTK_UNIT_POINTS);
+    double height = gtk_paper_size_get_height(paperSize, GTK_UNIT_POINTS);
+
+    switch (gtk_page_setup_get_orientation(m_pageSetup.get())) {
+    case GTK_PAGE_ORIENTATION_PORTRAIT:
+    case GTK_PAGE_ORIENTATION_REVERSE_PORTRAIT:
+        m_pageCanvas = recorder.beginRecording(width, height);
+        break;
+    case GTK_PAGE_ORIENTATION_LANDSCAPE:
+    case GTK_PAGE_ORIENTATION_REVERSE_LANDSCAPE:
+        m_pageCanvas = recorder.beginRecording(height, width);
+        break;
+    }
+
+    ASSERT(m_pageCanvas);
+}
+
+void WebPrintOperationGtk::endPage(SkPictureRecorder& recorder)
+{
+    ASSERT(m_pageCanvas);
+
+    if (currentPageIsLastPageOfSheet()) {
+        m_pages.append(recorder.finishRecordingAsPicture());
+        m_pageCanvas = nullptr;
+    }
+}
+
+void WebPrintOperationGtk::endPrint()
+{
+    SkDynamicMemoryWStream memoryBuffer;
+
+    auto document = SkPDF::MakeDocument(&memoryBuffer);
+    ASSERT(document);
+    for (auto page : m_pages) {
+        const auto& rect = page->cullRect();
+        auto canvas = document->beginPage(rect.width(), rect.height());
+        canvas->drawPicture(page);
+        document->endPage();
+    }
+    document->close();
+
+    memoryBuffer.flush();
+    sk_sp<SkData> pdfData = memoryBuffer.detachAsData();
+    ASSERT(pdfData);
+
+    printDone(WebCore::FragmentedSharedBuffer::create(pdfData.release()), { });
```

Comment:
I don't know why I made  `FragmentedSharedBuffer::create()` receive a pointer, but it should receive a `sk_sp<SkData>&&` so you could just do 
```
printDone(WebCore::FragmentedSharedBuffer::create(memoryBuffer.detachAsData()), { });
```

Reply:
Changed to that!